src/components/Block.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+
+interface TdsRadioButtonCustomEvent<T> extends Event {
+  target: EventTarget & T;
+}
+
+const Block: React.FC = () => {
+  const [modeVariant, setModeVariant] = useState('primary');
+
+	const capitalizeFirst = (str: string) => {
+		return str.charAt(0).toUpperCase() + str.slice(1);
+	};
+
+	const handleVariantChange = (event: TdsRadioButtonCustomEvent<{ radioId: string; value: string }>) => {
+    setModeVariant(event.target.value);
+  };
+
+  useEffect(() => {
+    const tdsRadioButtons = document.querySelectorAll('tds-radio-button');
+
+		const handleChange = (event: Event) => {
+      const target = event.target as HTMLElement;
+      const value = target.getAttribute('value');
+      if (value) {
+        setModeVariant(value);
+      }
+    };
+
+    tdsRadioButtons.forEach((tdsRadioButton) => {
+      tdsRadioButton.addEventListener('tdsChange', handleChange);
+
+      return () => {
+        tdsRadioButton.removeEventListener('tdsChange', handleChange);
+      };
+    });
+  }, []);
```

Comment:
I think these should be set on the element inside the runder function instead, in the way we show in our docs.